packages/sdk/src/protocols/base_protocol.ts

```diff
@@ -87,75 +77,85 @@ export class BaseProtocolSDK implements IBaseProtocolSDK {
       clearInterval(this.maintainPeersIntervalId);
       this.maintainPeersIntervalId = null;
       this.log.info("Maintain peers interval stopped");
+    } else {
+      this.log.debug("Maintain peers interval was not running");
     }
   }
 
   /**
-   * Checks if there are peers to send a message to.
-   * If `forceUseAllPeers` is `false` (default) and there are connected peers, returns `true`.
-   * If `forceUseAllPeers` is `true` or there are no connected peers, tries to find new peers from the ConnectionManager.
-   * If `autoRetry` is `false`, returns `false` if no peers are found.
-   * If `autoRetry` is `true`, tries to find new peers from the ConnectionManager with exponential backoff.
-   * Returns `true` if peers are found, `false` otherwise.
+   * Checks if there are sufficient peers to send a message to.
+   * If `forceUseAllPeers` is `false` (default), returns `true` if there are any connected peers.
+   * If `forceUseAllPeers` is `true`, attempts to connect to `numPeersToUse` peers.
    * @param options Optional options object
-   * @param options.autoRetry Optional flag to enable auto-retry with exponential backoff (default: false)
-   * @param options.forceUseAllPeers Optional flag to force using all available peers (default: false)
-   * @param options.initialDelay Optional initial delay in milliseconds for exponential backoff (default: 10)
-   * @param options.maxAttempts Optional maximum number of attempts for exponential backoff (default: 3)
-   * @param options.maxDelay Optional maximum delay in milliseconds for exponential backoff (default: 100)
+   * @param options.forceUseAllPeers Optional flag to force connecting to `numPeersToUse` peers (default: false)
+   * @param options.maxAttempts Optional maximum number of attempts to reach the required number of peers (default: 3)
+   * @returns `true` if the required number of peers are connected, `false` otherwise
    */
-  protected hasPeers = async (
+  protected async hasPeers(
     options: Partial<ProtocolUseOptions> = {}
-  ): Promise<boolean> => {
-    const {
-      autoRetry = false,
-      forceUseAllPeers = false,
-      initialDelay = 10,
-      maxAttempts = 3,
-      maxDelay = 100
-    } = options;
+  ): Promise<boolean> {
+    const { forceUseAllPeers = false, maxAttempts = 3 } = options;
+
+    this.log.debug(
+      `Checking for peers. forceUseAllPeers: ${forceUseAllPeers}, maxAttempts: ${maxAttempts}`
+    );
+
+    if (!forceUseAllPeers && this.connectedPeers.length > 0) {
+      this.log.debug(
+        `At least one peer connected (${this.connectedPeers.length}), not forcing use of all peers`
+      );
+      return true;
+    }
+
+    if (!forceUseAllPeers) {
```

Comment:
this `if` block and following `for` do almost similar things and I think `if` can be dropped

Reply:
you mean this? 1f42ed50a0481453cb0a8ea8fc099dfb18d03717